### DISCUSSION TITLE
Configuring for Next.js 13

### DISCUSSION THREAD

**Discussion by AlanGRutter** [heroui-inc/heroui#880] [Help]
Hi,
I am new to Next.js and NextUI and initally followed the guides to set up with Typescript and the pages folder, however I have been trying to do the same thing with the app folder used in the latest next.js version but I can't figure out what to put in the layout.tsx at the root level instead of _app.tsx and _document.tsx. The migration guide in next.js doesn't help me.

Is NextUI compatible with Next.js 13? If so, could someone please paste a snippet of what a basic layout using NextUIProvider would look like?  

Thanks
Alan

**Comment by irakli2206** [comment]
Currently what I do is wrap the body tag with NextThemesProvider and NextUIProvider, but the issue is that this gives Hydration error because those two providers are divs and div can't be a child of HTML. The app still works but it's all client rendered and really slow during development.
